tests/openebs-target-pod-failure_test.go

```diff
@@ -170,4 +201,93 @@ var _ = Describe("BDD of openebs target pod failure experiment", func() {
 		})
 	})
 
+	//Matching the Resource Verison after Chaos
+	Context("Check Resource Version of pool container", func() {
+
+		It("Should check for the change in Resource Version after Chaos", func() {
+			resourceVersionAfter := 0
+			csp_rv, err := client.CoreV1().Pods(cspPodNs).List(metav1.ListOptions{LabelSelector: cspPodLabels})
+			Expect(err).To(BeNil(), "fail to get the csp pods")
+			for _, podSpec := range csp_rv.Items {
+				resourceVersionAfter, _ = strconv.Atoi(podSpec.ResourceVersion)
+			}
+
+			Expect(resourceVersionAfter-resourceVersionBefore).NotTo(Equal(0), "The Resource Version does not change")
+			fmt.Println("The Resource Version changes")
+
+		})
+	})
+
+	//Matching the ContainerIDs after Chaos
+	Context("Check ContainerIDs after chaos", func() {
+
+		It("Should check for the change in ContainerIDs of csp pod", func() {
+
+			var containerIdAfter [3]string
+			containerCount := 0
+			containerIDChanged := false
+			csp, err := client.CoreV1().Pods(cspPodNs).List(metav1.ListOptions{LabelSelector: cspPodLabels})
+			Expect(err).To(BeNil(), "fail to get the csp pods")
+			for _, podSpec := range csp.Items {
+				for i := 0; i < len(podSpec.Status.ContainerStatuses); i++ {
+					containerIdAfter[containerCount] = (podSpec.Status.ContainerStatuses[i].ContainerID)
+					containerCount++
+				}
+			}
+
+			for i := range containerIdBefore {
+				if containerIdBefore[i] != containerIdAfter[i] {
+					containerIDChanged = true
+					break
+				}
+			}
+
+			Expect(containerIDChanged).NotTo(Equal("false"), "The Container ID does not change")
+			fmt.Println("Container ID Changes!!!")
+		})
+	})
+
+	//Matching the Container Restart Count after Chaos
+	Context("Check Container Restart Count", func() {
+
+		It("Should check for the change in Container Restart Count after Chaos", func() {
+
+			restartCountSumAfter := 0
+			csp_rc, err := client.CoreV1().Pods(cspPodNs).List(metav1.ListOptions{LabelSelector: cspPodLabels})
+			Expect(err).To(BeNil(), "fail to get the csp pods")
+			for _, podSpec := range csp_rc.Items {
+				for i := 0; i < len(podSpec.Status.ContainerStatuses); i++ {
+					restartCountSumAfter = restartCountSumAfter + int(podSpec.Status.ContainerStatuses[i].RestartCount)
+				}
+			}
+
+			Expect(restartCountSumAfter-restartCountSumBefore).NotTo(Equal(0), "The restart count does not change")
+			fmt.Println("The Restart count changes")
+
+		})
+	})
+
+	//Matching csp pod PodIP
+	Context("Check csp pod PodIP", func() {
+
+		It("Should check for the change in csp pod PodIP after Chaos", func() {
+
+			var podIpAfter string
+			podIpChanged := false
```

Comment:
Why you use this variable here? because podIP is not an array, you can directly compare